plugins/aea-ledger-cosmos/aea_ledger_cosmos/cosmos.py

```diff
@@ -783,6 +807,7 @@ def _get_storage_transaction(
         :param memo: any string comment.
         :return: the unsigned CosmWasm contract deploy message
         """
+        MsgStoreCode, Coin = (lazy_load()[k] for k in ("MsgStoreCode", "Coin"))
```

Comment:
I'm a bit unsure about this implementation but a small suggestion, instead of running for loop every time to extract required imports you can create a wrapper around the lazy_load method like this

```python
@functools.lru_cache()
 def lazy_load():
     ...

def load_modules(modules: Tuple[str,...]) -> Tuple[Any, ...]:
    "Docstring"
    imports = lazy_load()    
    return (imports[k] for k in modules)
    
MsgStoreCode, Coin = _import(("MsgStoreCode", "Coin"))
```

Reply:
just for sake of completeness, `functools.lru_cache` should do that, but it actually doesn't do a thing here, because python caches all imported modules already by default so